pom.xml

```diff
@@ -54,20 +60,21 @@
             <!-- TODO: should be transformed into a optional dependency -->
         </dependency>
         <dependency>
-            <groupId>org.jenkins-ci.plugins</groupId>
-            <artifactId>matrix-project</artifactId>
-            <!-- TODO: should be transformed into a optional dependency -->
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <!-- Preferred version is constrained above in dependencyManagement -->
         </dependency>
```

Comment:
I don't think you need to declare a direct dependency on this at all. It seems that the only reason you're dealing with `javax.annotation-api` at all was to make Enforcer happy. But Enforcer should be happy with the entry in `dependencyManagement` alone, without this addition. And this addition is not desirable because as I explained in my last comment it will pull in this dependency even if we don't directly consume it in the future.

```suggestion
```

Reply:
Ah, so those examples I saw with both dependencyManagement and dependency entries were an overkill of sorts? At least for this case? Checking if truly not required... :)

Reply:
Build and test pass :)